sphinx_immaterial/content_tabs.py

```diff
@@ -0,0 +1,182 @@
+    The ``MaterialTabSetHtmlTransform`` then transforms this container
+    into the HTML specific structure.
+    """
+
+    required_arguments = 1  # the tab label is the first argument
+    final_argument_whitespace = True
+    has_content = True
+    option_spec = {
+        "class": directives.class_option,
+    }
+
+    def run(self) -> List[nodes.Node]:
+        """Run the directive."""
+        self.assert_has_content()
+        if not is_md_tab_type(self.state_machine.node, "md-tab-set"):
+            LOGGER.warning(
+                "The parent of a 'md-tab-item' should be a 'md-tab-set'",
+                location=(self.env.docname, self.lineno),
+            )
+        tab_item = nodes.container(
+            "", is_div=True, type="md-tab-item", classes=["tabbed-block"]
+        )
+
+        # add tab label
+        textnodes, _ = self.state.inline_text(self.arguments[0], self.lineno)
+        tab_label = nodes.rubric(
+            self.arguments[0],
+            *textnodes,
+            classes=["tabbed-label"],
+        )
+        self.add_name(tab_label)
+        tab_item += tab_label
+
+        # add tab content
+        tab_content = nodes.container(
+            "",
+            is_div=True,
+            type="",
+            classes=["tabbed-block"] + self.options.get("class", []),
+        )
+        self.state.nested_parse(self.content, self.content_offset, tab_content)
+        tab_item += tab_content
+
+        return [tab_item]
+
+
+class content_tab_label(nodes.TextElement, nodes.General):
+    pass
+
+
+def visit_tab_label(self, node):
+    attributes = {"for": node["input_id"]}
+    self.body.append(self.starttag(node, "label", **attributes))
+
+
+def depart_tab_label(self, node):
+    self.body.append("</label>")
+
+
+def visit_tab_set(self: HTMLTranslator, node: content_tab_set):
+    # increment tab set counter
+    self.tab_set_count = getattr(self, "tab_set_count", 0) + 1
+
+    # configure tab set's div attributes
+    tab_set_identity = f"__tabbed_{self.tab_set_count}"
+    attributes = {"data-tabs": f"{self.tab_set_count}:{len(node.children)}"}
+    self.body.append(self.starttag(node, "div", **attributes))
+
+    # walkabout the children
+    tab_total = 0
+    tab_label_div = nodes.container("", is_div=True, classes=["tabbed-labels"])
+    tab_content_div = nodes.container("", is_div=True, classes=["tabbed-content"])
+    for tab_item in node.children:
+        try:
+            tab_label, tab_block = tab_item.children
+        except ValueError as exc:
+            raise ValueError(f"md-tab-item has no children:\n{repr(tab_item)}") from exc
+        tab_item_identity = tab_set_identity + f"_{tab_total + 1}"
+
+        # create: <input checked="checked" id="id" type="radio">
+        self.body.append(
+            "<input "
+            + ('checked="checked"' if not tab_total else "")
+            + f'type="radio" id="{tab_item_identity}" name="{tab_set_identity}">'
```

Comment:
Note: You mentioned you were concerned about the robustness --- it is true that properly escaping attributes is an important issue when generating HTML.  Here both of these attributes are generated by this extension themselves and are known not to need escaping.  However, to make it extra robust you could use `self.attval(tab_item_identity)` and `self.attval(tab_set_identity)` --- `attval` is a method implemented by docutils that is used by the `starttag` method to properly escape attribute values.

Reply:
thanks for the explanation! Adding it now.